test/Microsoft.ML.TestFramework/Microsoft.ML.TestFramework.csproj

```
@@ -1,6 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <DefineConstants>CORECLR</DefineConstants>
+    <DefineConstants Condition="'$(DefaultTestTargetFramework)' == 'net462'">CORECLR;NETFX</DefineConstants>
```

You might just want to append the DefineConstants and that will bring all the common constants for both frameworks, i.e, in this case CORECLR.
```suggestion
    <DefineConstants Condition="'$(DefaultTestTargetFramework)' == 'net462'">$(DefineConstants);NETFX</DefineConstants>
```

---

You don't need to define your own constant here. They are generated automatically by the build system.

See [https://docs.microsoft.com/en-us/dotnet/standard/frameworks](https://docs.microsoft.com/en-us/dotnet/standard/frameworks)

public class MyClass
{
    static void Main()
    {
#if NET40
        Console.WriteLine("Target framework: .NET Framework 4.0");
#elif NET45  
        Console.WriteLine("Target framework: .NET Framework 4.5");
#else
        Console.WriteLine("Target framework: .NET Standard 1.4");
#endif
    }
}

---
In reply to: [246516157](https://github.com/dotnet/machinelearning/pull/2018#discussion_r246516157) [](ancestors = 246516157)